openapi-v1.yaml

```diff
@@ -2682,6 +2692,118 @@ definitions:
         type: string
         description: output location of the exported file
 
+  TaskGraphLogStatus:
+    description: The status of a given task graph execution.
+    type: string
+    enum:
+      - unstarted
+        # The task graph has been submitted, but no tasks have been run.
+        # This should not be reported by the client.
+      - running
+        # Some task in the graph is currently running.
+        # This should not be reported by the client.
+      - idle
+        # The graph is not yet completed, but nothing is actively running.
+        # This should not be reported by the client.
+      - abandoned
+        # The graph has been idle for an extended period of time,
+        # likely due to crashing or aborting without notifying the server.
+        # Semantically this is equivalent to `idle`.
+      - succeeded
+        # The client has declared that the task graph has completed
+        # successfully. This ordinarily entails all nodes succeeding.
+      - failed
+        # The client has decided that the task graph was not successful.
+      - cancelled
+        # The client has cancelled further execution of the task graph.
+
+  TaskGraphLog:
+    description: Logging information about the execution of a task graph.
+    type: object
+    properties:
+      uuid:
+        type: string
+        readOnly: true
+        description: The server-generated UUID of the task graph.
+      namespace:
+        type: string
+        readOnly: true
+        description: The namespace that owns this task graph log.
+      created_by:
+        type: string
+        readOnly: true
+        description: The name of the user who created this task graph log.
+      start_time:
+        type: string
+        format: date-time
+        readOnly: true
+        x-omitempty: true
+        description: >
+          The start time of the task graph, recorded when the server
+          starts executing the first node.
+      end_time:
+        type: string
+        format: date-time
+        readOnly: true
+        x-omitempty: true
+        description: >
+          The end time of the task graph, recorded when the client reports
+          completion.
+      status:
+        $ref: '#/definitions/TaskGraphLogStatus'
+      nodes:
+        type: array
+        items:
+          $ref: '#/definitions/TaskGraphNodeMeta'
+        description: >
+          The structure of the graph. This is provided by the client
+          when first setting up the task graph. Thereafter, it is read-only.
+
+  TaskGraphNodeMeta:
+    description: Metadata about an individual node in a task graph.
+    type: object
+    properties:
+      client_node_uuid:
+        type: string
+        description: The client-generated UUID of the given graph node.
+      name:
+        type: string
+        description: >
+          The client-generated name of the node.
+          This is not guaranteed to be unique.
+      depends_on:
+        type: array
```

Comment:
This is a DAG, a node always depends on a single node. Why is this an array?

Reply:
A node could depend upon many nodes: task C takes the output of tasks A and B.